build.gradle.kts

```diff
@@ -0,0 +1,559 @@
+import com.diffplug.gradle.spotless.SpotlessExtension
+import com.google.protobuf.gradle.*
+import com.jfrog.bintray.gradle.BintrayExtension
+import com.jfrog.bintray.gradle.BintrayExtension.*
+import groovy.lang.GroovyObject
+import io.morethan.jmhreport.gradle.JmhReportExtension
+import me.champeau.gradle.JMHPluginExtension
+import nebula.plugin.release.git.opinion.Strategies
+import net.ltgt.gradle.errorprone.ErrorProneOptions
+import net.ltgt.gradle.errorprone.ErrorPronePlugin
+import org.gradle.api.plugins.JavaPlugin.*
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+import org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention
+import org.jfrog.gradle.plugin.artifactory.dsl.PublisherConfig
+import org.jfrog.gradle.plugin.artifactory.dsl.ResolverConfig
+import ru.vyarus.gradle.plugin.animalsniffer.AnimalSniffer
+import ru.vyarus.gradle.plugin.animalsniffer.AnimalSnifferPlugin
+
+plugins {
+    id("com.diffplug.spotless")
+    id("com.github.ben-manes.versions")
+    id("com.google.protobuf") apply false
+    id("com.jfrog.artifactory") apply false
+    id("com.jfrog.bintray") apply false
+    id("me.champeau.gradle.jmh") apply false
+    id("nebula.release")
+    id("net.ltgt.errorprone") apply false
+    id("ru.vyarus.animalsniffer") apply false
+    id("io.morethan.jmhreport") apply false
+}
+
+if (!JavaVersion.current().isJava11Compatible()) {
+    throw GradleException("JDK 11 or higher is required to build. " +
+            "One option is to download it from https://adoptopenjdk.net/. If you believe you already " +
+            "have it, please check that the JAVA_HOME environment variable is pointing at the " +
+            "JDK 11 installation.")
+}
+
+// Nebula plugin will not configure if .git doesn't exist, let's allow building on it by stubbing it
```

Comment:
how plausible will it be, now that we're using kotlin, to do some refactoring of the build script to make it more modular? Can we move pieces into files that are appropriately named to make it really easy to find the pieces related to publishing, for example?

Reply:
Yup I think we can do some splitting soon